### PR TITLE
Clean up OS X Build

### LINKED ISSUE #46: ReadME - Mac qt5.4.2 deprecated/EOL
The qt5.4.2 links toward the bottom of the ReadME are broken, due to 5.4.2's deprecation.

5.9.0 is out now.. will it still work?
https://download.qt.io/official_releases/qt/5.9/5.9.0/

### LINKED ISSUE #281: Old logo icon on mac-os.

### PR DESCRIPTION
**Pull request by ekryski**
In this PR:

- Properly link Boost so it's a 1-click install. Closes #46, #148, #150, #166, #182, #183, #198, #228 
- Remove macports configuration. We don't use that anymore
- Update OS X build instructions
- Update OS X usage. You no longer need to install boost 😄 
- Update to using c++11 on OS X
- Fix a bunch of c++11 warnings
- Ignore release `.zip` files so they're not checked into source control

This also resolves #281. New logo is in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,20 +84,12 @@ Binary (pre-compiled) wallets are available on all platforms at [https://vergecu
...
-    ```
-5. Double click the Verge-Qt application to open it.
-6. Go grab a :coffee: while it syncs with the blockchain
+4. Double click the Verge-Qt application to open it.
+5. Go grab a :coffee: while it syncs with the blockchain
 
 > **Note:** It may look like it is frozen or hung while it is indexing and syncing the blockchain. It's not. It's chugging away, but currently the UI doesn't give you a lot of feedback on status. We're working to fix that. Syncing takes a while to complete (ie. > 10 minutes or more) so just be patient.
 
-> **Note:** If you want to change your configuration the file is located at `~/Library/Application\ Support\VERGE\VERGE.conf`. This isn't required by default.
+> **Note:** If you want to change your configuration you can create a `~/Library/Application\ Support/VERGE/VERGE.conf` file with your options. If you don, make sure to restart Verge. This isn't required and so there is no `VERGE.conf` file created by default.
```
Comment (corymsmith) [inline README.md]: Small typo here, if you "don't" vs "don"

**Comment by justinvforvendetta** [comment]
this makes travis fail on seccomp for linux builds

**Comment by justinvforvendetta** [comment]
which is odd, because configure found it:
checking seccomp.h usability... yes
checking seccomp.h presence... yes
checking for seccomp.h... yes
checking for library containing seccomp_init... -lseccomp

**Comment by Positivism** [comment]
Confirmed working on high sierra though, try compiling it directly on nix, may be a travis issue.

File: src/wallet.cpp [hunk 2]
```diff
@@ -118,8 +118,6 @@ bool CWallet::Unlock(const SecureString& strWalletPassphrase)
 
 bool CWallet::ChangeWalletPassphrase(const SecureString& strOldWalletPassphrase, const SecureString& strNewWalletPassphrase)
 {
-    bool fWasLocked = IsLocked();
```
Comment (topcatx) [inline src/wallet.cpp]: intentional ? windows build went in with this change here..think this was reverted by @justinvforvendetta

Comment (ekryski) [reply src/wallet.cpp]: I had removed that because it is unused.

**Comment by ekryski** [comment]
Ya I dunno why Travis is bombing on this. @michpiatek was going to check the build on his machine but him and I were chatting last night and there isn't anything here that should affect linux or windows builds.

@justinvforvendetta maybe you can give it a go on your machine? I won't have to time to revisit this until later this week.

**Comment by michpiatek** [comment]
@ekryski @justinvforvendetta for some reason TARGET_OS check in configure.ac filtered out the PKG_CHECK_MODULES calls for libcap and libseccomp. Without those calls, LIBCAP_LIBS and LIBSECCOMP_LIBS variables were not set which gave linker errors.

However, I don't know why that check at configure.ac:555 didn't work on Travis. It works perfectly fine on my machine and it looks like it works fine with the current build running on Travis

**Pull request closed without merging**